### PR TITLE
Only compile wasm-bindgen when target_arch = "wasm32"

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
wasm-bindgen is an expensive dependency and should only be included when needed.

#### Summary of Changes
- Make all wasm_bindgen usage dependent on `target_arch = "wasm32"`.
- Remove `wasm_bindgen_stub` as the problem it solves goes away with this PR

`wasm_bindgen(skip)` doesn't work with `cfg_attr` so unfortunately we have to duplicate the definitions of the structs `Instruction`, `legacy::Message` and `Transaction`. I have checked if there is some other recommended workaround and there is not.

### CODE REVIEW

**Review by joncinque — COMMENTED** [review]
Thanks for the contribution!

It really stinks to have to repeat the struct definition, but I couldn't come up with a better alternative. I expanded the wasm macros and they add way more than just a second definition of the struct, so implementing the wasm traits by hand isn't a reasonable workaround.

In the wasm version of the structs, can you add a comment saying that the wasm version is required until https://github.com/rustwasm/wasm-bindgen/issues/3671 is addressed because `[wasm_bindgen(skip)]` doesn't work with `cfg_attr`? And that all changes between the two versions must be kept in sync?

File: sdk/src/signer/keypair.rs
Comment (joncinque) [inline sdk/src/signer/keypair.rs]: Sorry, maybe I'm missing something, but is this comment needed? There's only one version of Keypair because it doesn't require `#[wasm_bindgen(skip)]` anywhere

File: sdk/src/transaction/mod.rs
Comment (joncinque) [inline sdk/src/transaction/mod.rs:210]: Can you add the comment here?

File: sdk/program/src/pubkey.rs
Comment (joncinque) [inline sdk/program/src/pubkey.rs]: Is this comment needed? There's only one version of this because it doesn't require #[wasm_bindgen(skip)]

File: sdk/program/src/message/legacy.rs
Comment (joncinque) [inline sdk/program/src/message/legacy.rs:170]: Can you add the comment on top of this one?

File: sdk/program/src/instruction.rs
Comment (joncinque) [inline sdk/program/src/instruction.rs:350]: Can you add the comment on top of this one?

**Review by joncinque — COMMENTED** [review]
Just a couple of questions about the added comments, then this should be good to go

File: sdk/src/signer/keypair.rs
Comment (kevinheavey) [reply sdk/src/signer/keypair.rs]: ah woops

File: sdk/program/src/instruction.rs
Comment (joncinque) [inline sdk/program/src/instruction.rs]: Just this last one is missing:
```suggestion

/// wasm-bindgen version of the Instruction struct.
/// This duplication is required until https://github.com/rustwasm/wasm-bindgen/issues/3671
/// is fixed. This must not diverge from the regular non-wasm Instruction struct.
#[cfg(target_arch = "wasm32")]
#[wasm_bindgen]
pub struct Instruction {
```

**Review by joncinque — COMMENTED** [review]
Just one last change needed, but since it's just a comment, I'll commit it to the branch and get this moving through CI

**Review by joncinque — APPROVED** [review]
Looks great!

**Review by joncinque — COMMENTED** [review]
Ah darn, since I'm the last committer, we need one more approval before it goes through -- @buffalojoec when you have a moment, can I get a :heavy_check_mark:  please?

**Review by buffalojoec — APPROVED** [review]

**Comment by joncinque** [comment]
@kevinheavey sorry, can you do one last rebase to pick up the fix for the flaky CI?

**Pull request merged**
